catroid/src/org/catrobat/catroid/stage/StageActivity.java

```diff
@@ -115,6 +130,55 @@ public void onCreate(Bundle savedInstanceState) {
 		CameraManager.getInstance().setStageActivity(this);
 	}
 
+	private void setupAskHandler() {
+		messageHandler = new Handler(Looper.getMainLooper()) {
+			@Override
+			public void handleMessage(Message message) {
+				ArrayList<Object> params = (ArrayList<Object>) message.obj;
+
+				if (message.what == ASK_MESSAGE) {
+					showDialog((String) params.get(1), (AskAction) params.get(0));
+				}
+			}
+		};
+	}
+
+	private void showDialog(String question, final AskAction askAction) {
+		pause();
```

Comment:
Unfortunately this pause does not pause everything (for example vibration is not stopped. For whatever reason only StageActivity.pause() stops everything. I am not sure if this is intended behavior, so changing the StageListeners pause method may break other functionality.